doc/nrf/app_boards.rst

```diff
@@ -141,13 +142,42 @@ See :ref:`nrf_desktop_porting_guide` in the application documentation for detail
+The ``_ns`` suffix in the board target name refers to the adoption of the Security by Separation architecture principle, which splits the firmware to better protect sensitive assets and code.
+In Security by Separation, the firmware is stored in one of two security domains (partitions), either Secure Processing Environment (SPE) or Non-Secure Processing Environment (NSPE).
+This isolation of firmware is only possible if the underlying hardware supports it, but the concept is not limited to multi-core SoCs.
 
-* SPE stores all of the security features, including cryptography components and keys.
-* NSPE stores the main application, among other major components.
+Secure Processing Environment (SPE)
+===================================
 
-In Zephyr and the |NCS|, depending on the build target, you can build the firmware for the application core either for SPE alone (build targets without ``_ns``) or for both SPE and NSPE at the same time (build targets that include ``_ns``).
+SPE is present when you build for both the ``cpuapp`` and ``cpuapp_ns`` build targets.
+
+SPE with ``cpuapp_ns``
+----------------------
+
+In the Security by Separation scenario (``cpuapp_ns``), SPE implements security-critical functionality and data and isolates them from the application software (located in NSPE).
+The SPE environment contains secure firmware running in the secure state.
+
+The security of firmware is ensured with TrustZone, implemented through :ref:`Trusted Firmware-M (TF-M) <ug_tfm>`.
+TF-M integrates the `Platform Security Architecture (PSA)`_ API, which provides security features for the system, including roots of trust for protecting secrets, platform state, and cryptographic keys.
+The API also coordinates the communication with components in NSPE.
+
+SPE with ``cpuapp``
+-------------------
+
+In the scenario without separation of firmware (``cpuapp``), SPE is the only processing environment for the entire application core firmware.
+When you build for such targets, the firmware is placed only in the SPE, without TF-M.
+
+Non-Secure Processing Environment (NSPE)
+========================================
+
+NSPE implements the main user application and communication firmware, among other major components.
+
+SPE and NSPE in Zephyr and the |NCS|
+====================================
+
+In Zephyr and the |NCS|, SPE and NSPE are used when talking about the application core of a multi-core SoC, when the firmware is programmed to the ``cpuapp`` or ``cpuapp_ns`` build targets.
+Depending on the build target, you can build the firmware for the application core either for SPE alone (build targets without ``_ns``) or for both SPE and NSPE at the same time (build targets that include ``_ns``).
 
 On selecting the build target for NSPE (``_ns``), the application is built as a non-secure image, and an additional firmware component is built as the secure image.
```

Comment:
"On selecting the build target for NSPE" is contradicting the sentence just before a little bit, in my opinion. I think we might just write "On selecting the ``_ns`` build target" and leave the rest of the sentence as is. Or:
```suggestion
On selecting the ``_ns`` build target, the application is built as a non-secure image to reside in NSPE, and an additional firmware component is built as the secure image to reside in SPE.
```

Reply:
This section was re-written.

Reply:
Perfect!